src/component/modal/modal.tsx

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+class Modal extends React.PureComponent{
+  public render = () =>
+    ReactDOM.createPortal(
+      this.props.children,
+      document.getElementById('modal-container') as Element
+    )
+}
+
+export default Modal;
```

Comment:
Modal is a React Pure Component and it will render its children inside the 'modal-container' using React 16s Portal.